features/FEATURE_LWIP/lwip-interface/ppp_lwip.cpp

```diff
@@ -373,6 +372,9 @@ nsapi_error_t nsapi_ppp_connect(FileHandle *stream, Callback<void(nsapi_event_t,
         retcode = lwip._add_ppp_interface(stream, true, stack, &my_interface);
         if (retcode != NSAPI_ERROR_OK) {
             my_interface = NULL;
+            my_stream->set_blocking(true);
+            my_stream = NULL;
+            connection_status_cb = NULL;
```

Comment:
AT_CellularNetwork::open_data_channel should release file handle before calling nsapi_ppp_connect(), and also reinitialise file handle if nsapi_ppp_connect() returns failure.

Reply:
By "release" do you mean unset its `sigio` callback? Yes, probably wise.

Reply:
no need to do it here, only after my_interface->bringup which will set sigio. Added unset of sigio there. Good point!